py/picca/pk1d/postproc_pk1d.py

```diff
@@ -820,15 +824,20 @@ def compute_average_pk_redshift(
                 variance_estimated_col = fitfunc_variance_pk1d(data_snr, *coef_col)
                 weights_col = 1.0 / variance_estimated_col
                 if apply_z_weights:
-                    weights *= redshift_weights
-                mean = np.average(data_values, weights=weights)
+                    mean = np.average(data_values, weights=weights * redshift_weights)
+                else:
+                    mean = np.average(data_values, weights=weights)
                 if apply_z_weights:
                     # Analytic expression for the re-weighted average:
                     error = np.sqrt(np.sum(weights_col * redshift_weights)) / np.sum(
                         weights_col
                     )
                 else:
                     error = np.sqrt(1.0 / np.sum(weights_col))
+                    # Variance estimator derived by Jean-Marc, we keep the estimated one.
```

Comment:
are we now using Jean-Marc's estimator here, and is that just the same as before? Or are we using something else?

Reply:
This was just a test and I saved it, but we keep the same as before for now. Honestly it does not change a lot the level of the statistical error.

Reply:
so the estimator by JM (on which we base the covar output as well) is different from the estimator actually used. I think this is sub-optimal and we might want to use JMs in the future. We'd definitely need to document the estimator actually used in the papers

Reply:
Yes the two estimators are differents but I am renormalizing the covariance matrix so that the diagonal is the same than our estimator. It mainly impact very small scales, e.g. on a small test example: 
![image](https://github.com/user-attachments/assets/27bfbcd2-4253-4f65-a647-41fef20624f2)

Reply:
Right now we are cutting k < 2.0 AA^-1